spec/models/game_spec.rb

```diff
@@ -74,4 +74,61 @@
       expect(game.active_player).to eq(game.black_player)
     end
   end
+
+  describe '#stalemate?' do
+    before do
+      allow_any_instance_of(Game).to receive(:populate_board!).and_return true
+    end
+
+    let(:game) { FactoryGirl.create(:game) }
+    let(:white_king) { FactoryGirl.create(:king, type: 'King', color: 'white', row: 7, column: 3) }
+    let(:black_king) { FactoryGirl.create(:king, type: 'King', color: 'black', row: 0, column: 3) }
+
+    it 'should return true if active player is in stalemate' do
+      game.pieces << white_king
+      game.pieces << black_king
+      game.pieces << FactoryGirl.create(:black_pawn, type: 'Pawn', row: 6, column: 3)
+      game.pieces << FactoryGirl.create(:white_pawn, type: 'Pawn', row: 1, column: 3)
```

Comment:
I'm not sure if this is actually setting up a stalemate scenario, I could definitely be wrong though but might want to just double check. Can't the king just capture the pawn in front of it? Again, I'm definitely no expert at chess so if I'm wrong just go ahead and ignore this comment :smile: